code/__HELPERS/_lists.dm

```diff
@@ -528,6 +528,18 @@
 		result = pick_weight(fill_with_ones(result))
 	return result
 
+/**
+* Like pick_weight, but decreases the value of the picked element by 1
+ * For example, given the following list:
+ * A = 6, B = 3, C = 1, D = 0
+ * A would have a 60% chance of being picked, after which it would decrease by one and the new list would be
+ * A = 5, B = 3, C = 1, D = 0
+ * Tt would then have a 55.55...% to be picked, rinse and repeat
+*/
+/proc/pick_weight_take(list/list_to_pick)
+	. = pick_weight(list_to_pick)
+	list_to_pick[.]--
```

Comment:
this is a bit... weird, makes actual values matter instead of proportion

Reply:
I'm not sure I get you? It's so I can define quantities of available options and draw from those without getting duplicates if I don't want to

Reply:
we normally act like weighted lists only matter in so far as the weights are in relation to the sum, right?
so like, if you de/increase weights by some factor it doesn't change behavior, and this changes this in some cases, s all.

Reply:
Oh yeah I want it to change the chances, otherwise I'd just have grabbed the normal pickweight